client/taskenv/env.go

```diff
@@ -603,7 +603,6 @@ func (b *Builder) setAlloc(alloc *structs.Allocation) *Builder {
 
 	tg := alloc.Job.LookupTaskGroup(alloc.TaskGroup)
 
-	// COMPAT(0.11): Remove in 0.11
 	b.otherPorts = make(map[string]string, len(tg.Tasks)*2)
 	if alloc.AllocatedResources != nil {
```

Comment:
I kept this conditional because `TestClient_AddAllocError` asserts that when an invalid alloc is passed, taskenv doesn't panic and NewTaskRunner returns an error.  Not sure what the conditions the test actually tests for.